xds/internal/xdsclient/clientimpl.go

```diff
@@ -19,48 +19,124 @@
 // is a wrapper of this struct with a ref count.
 type clientImpl struct {
-	done               *grpcsync.Event
-	config             *bootstrap.Config
-	logger             *grpclog.PrefixLogger
-	watchExpiryTimeout time.Duration
-	backoff            func(int) time.Duration // Backoff for ADS and LRS stream failures.
-	serializer         *grpcsync.CallbackSerializer
-	serializerClose    func()
-	resourceTypes      *resourceTypeRegistry
-
-	// authorityMu protects the authority fields. It's necessary because an
-	// authority is created when it's used.
-	authorityMu sync.Mutex
-	// authorities is a map from ServerConfig to authority. So that
-	// different authorities sharing the same ServerConfig can share the
-	// authority.
-	//
-	// The key is **ServerConfig.String()**, not the authority name.
-	//
-	// An authority is either in authorities, or idleAuthorities,
-	// never both.
-	authorities map[string]*authority
-	// idleAuthorities keeps the authorities that are not used (the last
-	// watch on it was canceled). They are kept in the cache and will be deleted
-	// after a timeout. The key is ServerConfig.String().
+	// The following fields are initialized at creation time and are read-only
+	// after that, and therefore can be accessed without a mutex.
+	done               *grpcsync.Event              // Fired when the client is closed.
+	topLevelAuthority  *authority                   // The top-level authority, used only for old-style names without an authority.
+	authorities        map[string]*authority        // Map from authority names in bootstrap to authority struct.
+	config             *bootstrap.Config            // Complete bootstrap configuration.
+	watchExpiryTimeout time.Duration                // Expiry timeout for ADS watch.
+	backoff            func(int) time.Duration      // Backoff for ADS and LRS stream failures.
+	transportBuilder   transport.Builder            // Builder to create transports to the xDS server.
+	resourceTypes      *resourceTypeRegistry        // Registry of resource types, for parsing incoming ADS responses.
+	serializer         *grpcsync.CallbackSerializer // Serializer for invoking watcher callbacks.
+	serializerClose    func()                       // Function to close the serializer.
+	logger             *grpclog.PrefixLogger        // Logger for this client.
+
+	// The clientImpl owns a bunch of channels to individual xDS servers
+	// specified in the bootstrap configuration. Authorities acquire references
+	// to these channels based on server configs within the authority config.
+	// The clientImpl maintains a list of interested authorities for each of
+	// these channels, and forwards updates from the channels to each of these
+	// authorities.
 	//
-	// An authority is either in authorities, or idleAuthorities,
-	// never both.
-	idleAuthorities *cache.TimeoutCache
+	// Once all references to a channel are dropped, the channel is moved to the
```

Comment:
so according to this XdsClient will not behave as a regular grpc client which needs to be closed by the user of xds client?